src/systems/systems/src/pressurization/mod.rs

```diff
@@ -0,0 +1,815 @@
+use self::{cabin_pressure_controller::CabinPressureController, pressure_valve::PressureValve};
+use crate::{
+    shared::{random_number, EngineCorrectedN1},
+    simulation::{
+        Read, SimulationElement, SimulationElementVisitor, SimulatorReader, SimulatorWriter,
+        UpdateContext, Write,
+    },
+};
+use uom::si::{f64::*, length::foot, pressure::hectopascal, velocity::foot_per_minute};
+
+mod cabin_pressure_controller;
+mod pressure_valve;
+
+trait PressureValveActuator {
+    fn should_open_pressure_valve(&self) -> bool;
+    fn should_close_pressure_valve(&self) -> bool;
+    fn target_valve_position(&self) -> Ratio;
+}
+
+pub struct Pressurization {
+    cpc: [CabinPressureController; 2],
+    outflow_valve: PressureValve,
+    active_system: usize,
+    landing_elevation: Length,
+    sea_level_pressure: Pressure,
+    destination_qnh: Pressure,
+}
+
+impl Pressurization {
+    pub fn new() -> Self {
+        let random = random_number();
+        let mut active: usize = 1;
+        if random % 2 == 0 {
+            active = 2
+        }
+
+        Self {
+            cpc: [CabinPressureController::new(); 2],
+            outflow_valve: PressureValve::new(),
+            active_system: active,
+            landing_elevation: Length::new::<foot>(0.),
+            sea_level_pressure: Pressure::new::<hectopascal>(1013.25),
+            destination_qnh: Pressure::new::<hectopascal>(0.),
+        }
+    }
+
+    pub fn update(&mut self, context: &UpdateContext, engines: [&impl EngineCorrectedN1; 2]) {
+        for c in self.cpc.iter_mut() {
+            c.update(
+                context,
+                engines,
+                self.landing_elevation,
+                self.sea_level_pressure,
+                self.destination_qnh,
+            );
+            self.outflow_valve.update(context, c);
+        }
+        self.switch_active_system();
+    }
+
+    fn switch_active_system(&mut self) {
+        let mut changed = false;
+        for c in &mut self.cpc {
+            if c.should_switch_cpc() {
+                if !changed {
+                    self.active_system = if self.active_system == 1 { 2 } else { 1 };
+                    changed = true;
+                }
+                c.reset_cpc_switch();
+            }
+        }
```

Comment:
I'll be honest. I'm still not sure why each individual controller is deciding if there should be switching. But this might be the reality 😀. In any case I'd suggest writing this slightly different, to remove the need for the `changed` variable.

```suggestion
        if self
            .cpc
            .iter()
            .any(|controller| controller.should_switch_cpc())
        {
            self.active_system = if self.active_system == 1 { 2 } else { 1 };
        }

        self.cpc
            .iter_mut()
            .filter(|controller| controller.should_switch_cpc())
            .for_each(|controller| {
                controller.reset_cpc_switch();
            });
```

Reply:
Really nice, applied!